modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/IgniteH2Indexing.java

```diff
@@ -1114,6 +1117,17 @@ private void checkClusterState(QueryParserResult parseRes) {
                     // Check if cluster state is valid.
                     checkClusterState(parseRes);
 
+                    if (ctx.event().isRecordable(EVT_QUERY_EXECUTION)) {
+                        ctx.event().record(new QueryExecutionEvent<>(
+                            ctx.discovery().localNode(),
+                            CacheQueryType.SQL_FIELDS.name() + " query execution.",
+                            EVT_QUERY_EXECUTION,
+                            CacheQueryType.SQL_FIELDS.name(),
+                            newQryDesc.sql(),
+                            newQryParams.arguments(),
+                            ctx.localNodeId()));
```

Comment:
AFAIK this is not correct `secSubjId`.
What if SQL query executed from thin client?

Reply:
Fixed.